### ISSUE TITLE
Missing *.js in tmp folder created by now dev

### ISSUE THREAD

**Issue by shaunmolloy** [vercel/vercel#2378]
Noticed an issue with `now dev` and the deployment.

I generate a static build of Gatsby into a public folder.
Then symlink the public folder into an express folder.

Example of structure:

    └── packages
        └── gatsby
            ├── public
        ├── express
        │   └── public -> ../gatsby/public
        │   └── index.js

I noticed the problem as I see a lot of 404s.

<img width="416" alt="Screenshot 2019-05-24 at 19 47 03" src="https://user-images.githubusercontent.com/8168925/58350035-bf1dc100-7e5c-11e9-844c-77716edfb8ba.png">

> express/public/
> **30 files**

<img width="722" alt="Screenshot 2019-05-24 at 19 33 04" src="https://user-images.githubusercontent.com/8168925/58350471-e1fca500-7e5d-11e9-82d7-22e63b19c0bb.png">

> /private/var/folders/.../packages/express/public
> **20 files**

<img width="718" alt="Screenshot 2019-05-24 at 19 32 11" src="https://user-images.githubusercontent.com/8168925/58350616-4ddf0d80-7e5e-11e9-83c4-45d477f23a29.png">

**10 files missing from tmp folder:**
- 1-38089c36d96455b905b0.js
- app-e5b8ef0059cba8c01b9c.js
- component---node-modules-gatsby-plugin-offline-app-shell-js-896a3f852b233b9ca05e.js
- component---src-pages-404-js-9a82b75f659524394002.js
- component---src-templates-page-js-f630f132f289fd2fd011.js
- idb-keyval-iife.min.js
- pages-manifest-5bb70dd5f5aa579dcde2.js
- styles-b76dfad3fcff2976e17c.js
- sw.js
- webpack-runtime-0d693ef5154d2d09c053.js

Oddly all JS files.

This affects stable and canary versions of now-cli.

**Comment by shaunmolloy** [comment]
I thought the symlink caused this, but apparently not.
Attempted to move away from the workspace.
I put both gatsby and express in the same folder, but notice the same issue.

**Comment by TooTallNate** [comment]
Hey there, thanks for the report. Could you create a GitHub repo that demonstrates the issue so that we can take a closer look? Thanks in advance!

**Comment by shaunmolloy** [comment]
Hey @TooTallNate,

I've created the repo here.
https://github.com/shaunmolloy/zeit-gatsby-express

Run `yarn`.
Then call `now dev`.

After about 30s, you can fire up http://localhost:3000/
If you open dev tools and look at the network pane, you'll notice a bunch of 404s.

**Comment by krasimir** [comment]
Hey @shaunmolloy,

You rely on the `@now/node-server` builder. That builder uses [ncc](https://github.com/zeit/ncc) to compile your app and sends it to the [fun](https://github.com/zeit/fun)'s lambda runner. If you add

```js
app.use(function(req, res, next) {
  console.log(publicPath);
  next();
});
```
in your `express/index.js` you'll see a path printed to the console. Something like:

```
/private/var/folders/lm/cncl8rm52n516j5m54qwbb540000gn/T/zeit-fun-fef1a511c48ee/packages/express/public
```
(That's the case if you use iOS).

If you explore that folder you'll see that the `*.js` files are missing there. So, they are not even send to the lambda runner (fun). The problem is in the ncc compiler and how it _compiles_ your project. The current 0.18.5 version does not include your `*.js` files into the list of assets. The newest release `0.19.0-beta.1` however does that. You can verify it by creating (for example) `test_ncc.js` file with the following content (in your root project dir):

```js
const ncc = require(`@zeit/ncc`);

ncc(`${ __dirname }/packages/express/index.js`, {
  sourceMap: true,
  sourceMapRegister: true,
}).then(({ code, map, assets }) => {
  console.log(Object.keys(assets));
});
```

Then run these two commands:

```js
> npm i @zeit/ncc
> node ./test_ncc.js
```

You'll see all of your files listed. ncc relies on webpack for _compiling_ your project and there is a `@zeit/webpack-asset-relocator-loader` loader which is updated in the `0.19.0-beta.1` version and I think that's the actual fix of your problem.

So, shortly :) someone from Zeit team should update the `@now/node-server` to use the latest ncc version. You can't trick it by installing it on your project because the now builders are reinstalled on every `now dev` run. They live in `/Users/<user>/Library/Caches/co.zeit.now/dev/builders` which is a location that has nothing to do with your `node_modules` folder.

cc @TooTallNate @guybedford

P.S.
The commit that I think will fix this issue https://github.com/zeit/ncc/commit/567d25961d428705f2aa8ab695e01432313ba830#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R26

**Comment by shaunmolloy** [comment]
Hi all,

I take it we'd really be waiting on the next stable release
for ncc, rather than using a beta version.
https://github.com/zeit/ncc/releases

Which would be when `0.19.0-beta.1` becomes `0.19.0`.

Then the ncc version can be updated in node-server:
https://github.com/zeit/now-builders/blob/canary/packages/now-node-server/package.json

Let me know if there's any timeline on the release.
Thanks!

**Comment by TooTallNate** [comment]
Closing because `ncc` is no longer used in the `@now/node` builder (now `node-file-trace` is used).

Please try again using the latest `@now/node` builder, and create a new issue if you have any troubles.
